File: day1/day1.py
# part 1
max_index = max_cals = 0
elves = [0]
for line in open("input.txt"):
    l = line.strip()
    if l.isnumeric():
        elves[-1] += int(l)
    else:
        elves.append(0)

    if elves[-1] > max_cals:
        max_cals = elves[-1]
        max_index = len(elves) - 1
print("Part 1", max_cals)

# part 2
elves = [0]
for line in open("input.txt"):
    l = line.strip()
    if l.isnumeric():
        elves[-1] += int(l)
    else:
        elves.append(0)
elves = sorted(elves)
print("Part 2", sum(elves[-3:]))

File: day1/day1.rs
use std::fs::File;
use std::io::{self, BufRead};
use std::path::Path;

fn main() {
    if let Ok(lines) = read_lines("./input.txt") {
        let mut elves = vec![0];
        for line in lines {
            if let Ok(cals) = line.unwrap().parse::<usize>() {
                *elves.last_mut().unwrap() += cals;
            } else {
                elves.push(0);
            }
        }
        elves.sort();
        println!("Part 1: {:?}", elves.last().unwrap());
        println!("Part 2: {:?}", elves.iter().rev().take(3).sum::<usize>());
    }
}

// from rust by example
// https://doc.rust-lang.org/rust-by-example/std_misc/file/read_lines.html
fn read_lines<P>(filename: P) -> io::Result<io::Lines<io::BufReader<File>>>
where P: AsRef<Path>, {
    let file = File::open(filename)?;
    Ok(io::BufReader::new(file).lines())
}
